db/db_impl.cc

```diff
@@ -2296,15 +2296,17 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
       return Status::OK();
```

Comment:
If cfd->imm()->NumNotFlushed() != 0 && cfd->mem()->IsEmpty(), then memtable will still be switched (even if it is empty), which may cause a round of unnecessary flush job if this memtable is the only one to be flushed in the flush job.
This case is caused when some immutable memtables are being flushed in the background, and the current memtable is empty, and the Flush() is invoked.
Of course, this case is quite infrequent in normal usage, and will not cause error. I meet it when I want to do checkpoint periodically in background thread.